src/Stratis.Bitcoin.Features.Miner/BlockDefinition.cs

```diff
@@ -367,61 +365,65 @@ protected virtual void AddTransactions(out int nPackagesSelected, out int nDesce
                 if (mi == null)
                 {
                     modit = mapModifiedTx.Values.OrderBy(o => o, compare).First();
-                    iter = modit.MempoolEntry;
+                    mempoolEntry = modit.MempoolEntry;
                     fUsingModified = true;
                 }
                 else
                 {
                     // Try to compare the mapTx entry to the mapModifiedTx entry
-                    iter = mi;
+                    mempoolEntry = mi;
 
                     modit = mapModifiedTx.Values.OrderBy(o => o, compare).FirstOrDefault();
-                    if ((modit != null) && (compare.Compare(modit, new TxMemPoolModifiedEntry(iter)) < 0))
+                    if ((modit != null) && (compare.Compare(modit, new TxMemPoolModifiedEntry(mempoolEntry)) < 0))
                     {
                         // The best entry in mapModifiedTx has higher score
                         // than the one from mapTx..
                         // Switch which transaction (package) to consider.
 
-                        iter = modit.MempoolEntry;
+                        mempoolEntry = modit.MempoolEntry;
                         fUsingModified = true;
                     }
                     else
                     {
                         // Either no entry in mapModifiedTx, or it's worse than mapTx.
                         // Increment mi for the next loop iteration.
-                        ancestorScoreList.Remove(iter);
+                        ancestorScoreList.Remove(mempoolEntry);
                     }
                 }
 
                 // We skip mapTx entries that are inBlock, and mapModifiedTx shouldn't
                 // contain anything that is inBlock.
-                Guard.Assert(!this.inBlock.Contains(iter));
+                Guard.Assert(!this.inBlock.Contains(mempoolEntry));
 
-                long packageSize = iter.SizeWithAncestors;
-                Money packageFees = iter.ModFeesWithAncestors;
-                long packageSigOpsCost = iter.SigOpCostWithAncestors;
+                long packageSize = mempoolEntry.SizeWithAncestors;
+                Money packageFees = mempoolEntry.ModFeesWithAncestors;
+                long packageSigOpsCost = mempoolEntry.SigOpCostWithAncestors;
                 if (fUsingModified)
                 {
                     packageSize = modit.SizeWithAncestors;
                     packageFees = modit.ModFeesWithAncestors;
                     packageSigOpsCost = modit.SigOpCostWithAncestors;
                 }
 
-                if (packageFees < this.BlockMinFeeRate.GetFee((int)packageSize))
+                // Don't check the package fees if this is a CirrusRewardScript transaction.
```

Comment:
I note that this would only be considering transactions that have already got through the mempool's rules. So the blanket bypass is ok as the mempool's rules are much stricter about the transaction format that is exempt from fees.